### PR TITLE
Modify Core deployment for chain expansion

### PR DESCRIPTION
**Pull request by nambrot**
This PR modifies core deployment scripts to make use of https://github.com/abacus-network/abacus-monorepo/pull/1019

It skips recovery for `test` environment.

It also fixes a few places where we don't properly calculating the subset of config and multiprovider chains

### CODE REVIEW

File: typescript/infra/scripts/core.ts [hunk 1]
```diff
@@ -13,54 +16,73 @@ import {
...
-  getEnvironmentDirectory,
 } from './utils';
 
+function mergeVerificationInputs<ChainName extends Chains>(
+  existingInputsMap: ChainMap<ChainName, ContractVerificationInput[]>,
+  newInputsMap: ChainMap<ChainName, ContractVerificationInput[]>,
+): ChainMap<ChainName, ContractVerificationInput[]> {
+  const allChains = new Set<ChainName>();
+  Object.keys(existingInputsMap).forEach((_) => allChains.add(_ as ChainName));
+  Object.keys(newInputsMap).forEach((_) => allChains.add(_ as ChainName));
```
Comment (yorhodes) [inline typescript/infra/scripts/core.ts:27]: in hindsight not sure we should merge these
they are not appropriately structured to be merged

File: typescript/infra/src/config/agent.ts [hunk 2]
```diff
@@ -242,13 +242,13 @@ export type RustContractBlock<T> = {
 };
 
 export type OutboxAddresses = {
-  outbox: types.Address;
+  outbox?: types.Address;
   interchainGasPaymaster?: types.Address;
 };
```
Comment (yorhodes) [inline typescript/infra/src/config/agent.ts:247]: I wouldn't change these types, and instead use `Partial` elsewhere

File: typescript/infra/src/core/deploy.ts [hunk 3]
```diff
@@ -28,8 +25,8 @@ export class AbacusCoreInfraDeployer<
         inboxes: {},
         outbox: {
           addresses: {
-            outbox: contracts.outbox.address,
-            interchainGasPaymaster: contracts.interchainGasPaymaster.address,
+            outbox: contracts?.outbox?.address,
+            interchainGasPaymaster: contracts?.interchainGasPaymaster?.address,
```
Comment (yorhodes) [inline typescript/infra/src/core/deploy.ts:29]: can we just make `deployedContracts` a `Partial<Contracts>`

File: typescript/sdk/src/deploy/core/AbacusCoreDeployer.ts [hunk 4]
```diff
@@ -156,7 +156,11 @@ export class AbacusCoreDeployer<Chain extends ChainName> extends AbacusDeployer<
       }
     });
 
-    const remotes = this.multiProvider.remoteChains(chain);
+    const configChains = Object.keys(this.configMap) as Chain[];
+    const remotes = this.multiProvider
+      .remoteChains(chain)
+      .filter((_) => configChains.includes(_));
+
```
Comment (yorhodes) [inline typescript/sdk/src/deploy/core/AbacusCoreDeployer.ts:163]: in hindsight I think config being larger than multiprovider should throw

**Review by yorhodes — COMMENTED** [review]
some questions about approach

File: typescript/sdk/src/deploy/core/AbacusCoreDeployer.ts (hunk 4 above)
Comment (nambrot) [reply typescript/sdk/src/deploy/core/AbacusCoreDeployer.ts:163]: Yeah, that was originally also my thought

File: typescript/infra/src/core/deploy.ts (hunk 3 above)
Comment (nambrot) [reply typescript/infra/src/core/deploy.ts:29]: it is no?

**Review by yorhodes — APPROVED** [review]

**Pull request merged**